### DISCUSSION TITLE
How to darkMode without adding dark:bg-primary-dark to elements

### DISCUSSION THREAD

**Discussion by kostyakazmiruk** [tailwindlabs/tailwindcss#10457] [Help] [answered]
I am using Nextjs 12, ReactJs 18.2, tailwindcss 3.1.7

I saw the example in documentation, 
<html class="dark">
<body>
  
  <div class="bg-white dark:bg-black">
    
  </div>
</body>
</html>

But was wondering for a while:
Is there any way, not to use darkClass for enabling darkMode for the page. 
Like in the example above, to each element, we need to pass 

<div class="bg-white dark:bg-black"> or
<div class="bg-white dark:bg-black hover:bg-white">

Can we create a Context in React that will deal with dark mode and just add suffix to our CSS variable:
<div class={"w-24 border shadow rounded" + {"bg-white-${darkmode}"}}>
colors: {
        primary: {
          DEFAULT: "#0078B4",
          dark: "#006699",
           hover: "#FFFFFF",
        },
}

Or it is just a bad idea?

**Comment by wongjn** [comment, marked as answer]
Here is a [maintainer's response to a similar question](https://github.com/tailwindlabs/tailwindcss/discussions/10274#discussioncomment-4627634):
> I would define all of my colors as CSS variables and use a parent selector just to change the values of the variables, no need to create tons of custom classes:
> 
> https://tailwindcss.com/docs/customizing-colors#using-css-variables
> 
> This demo is several versions behind so some things are out of date but should communicate the idea clearly at least:
> 
> https://github.com/adamwathan/theming-tailwind-demo
> 
> Personally I think doing dark mode this way is seductive but not as practical as you expect. Tons of situations where you end up swapping the same color in light mode for two different colors in dark mode depending on how it’s used, and then you need to come up with different names for those two colors and it quickly becomes a very big project.
